php_phongo.c

```diff
@@ -1777,140 +1778,146 @@ static bool php_phongo_apply_rp_options_to_uri(mongoc_uri_t* uri, bson_t* option
+			if (max_staleness_seconds != MONGOC_NO_MAX_STALENESS) {
 
-			return false;
-		}
+				if (max_staleness_seconds < MONGOC_SMALLEST_MAX_STALENESS_SECONDS) {
+					phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected maxStalenessSeconds to be >= %d, %" PRId64 " given", MONGOC_SMALLEST_MAX_STALENESS_SECONDS, max_staleness_seconds);
+					mongoc_read_prefs_destroy(new_rp);
 
-		max_staleness_seconds = bson_iter_as_int64(&iter);
+					return false;
+				}
 
-		if (max_staleness_seconds != MONGOC_NO_MAX_STALENESS) {
+				if (max_staleness_seconds > INT32_MAX) {
+					phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected maxStalenessSeconds to be <= %d, %" PRId64 " given", INT32_MAX, max_staleness_seconds);
+					mongoc_read_prefs_destroy(new_rp);
 
-			if (max_staleness_seconds < MONGOC_SMALLEST_MAX_STALENESS_SECONDS) {
-				phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected maxStalenessSeconds to be >= %d, %" PRId64 " given", MONGOC_SMALLEST_MAX_STALENESS_SECONDS, max_staleness_seconds);
-				mongoc_read_prefs_destroy(new_rp);
+					return false;
+				}
 
-				return false;
+				if (mongoc_read_prefs_get_mode(new_rp) == MONGOC_READ_PRIMARY) {
+					phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Primary read preference mode conflicts with maxStalenessSeconds");
+					mongoc_read_prefs_destroy(new_rp);
+
+					return false;
+				}
 			}
 
-			if (max_staleness_seconds > INT32_MAX) {
-				phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected maxStalenessSeconds to be <= %d, %" PRId64 " given", INT32_MAX, max_staleness_seconds);
-				mongoc_read_prefs_destroy(new_rp);
+			mongoc_read_prefs_set_max_staleness_seconds(new_rp, max_staleness_seconds);
+		}
+	}
 
-				return false;
-			}
+	if (mongoc_read_prefs_get_mode(new_rp) == MONGOC_READ_PRIMARY &&
+		!bson_empty(mongoc_read_prefs_get_tags(new_rp))) {
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Primary read preference mode conflicts with tags");
+		mongoc_read_prefs_destroy(new_rp);
 
-			if (mongoc_read_prefs_get_mode(new_rp) == MONGOC_READ_PRIMARY) {
-				phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Primary read preference mode conflicts with maxStalenessSeconds");
-				mongoc_read_prefs_destroy(new_rp);
+		return false;
+	}
 
-				return false;
-			}
-		}
+	/* Make sure maxStalenessSeconds is not combined with primary readPreference */
+	if (mongoc_read_prefs_get_mode(new_rp) == MONGOC_READ_PRIMARY &&
+		mongoc_read_prefs_get_max_staleness_seconds(new_rp) != MONGOC_NO_MAX_STALENESS) {
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Primary read preference mode conflicts with maxStalenessSeconds");
+		mongoc_read_prefs_destroy(new_rp);
 
-		mongoc_read_prefs_set_max_staleness_seconds(new_rp, max_staleness_seconds);
+		return false;
 	}
 
 	/* This may be redundant in light of the last check (primary with tags), but
```

Comment:
While we're modifying this function, can we update this comment to "in light of the previous checks (primary with tags or maxStalenessSeconds)"?

Reply:
Updated the comment as per your suggestion.